astropy/visualization/wcsaxes/tests/test_frame.py

```diff
@@ -72,8 +72,6 @@ def test_custom_frame(self):
         # Clip the image to the frame
         im.set_clip_path(ax.coords.frame.patch)
 
-        return fig
```

Comment:
Are you sure this isn't needed for them weird hash image comparison that CircleCI picks up?

Reply:
I think this is intentional... #5496 cc @astrofrog

Reply:
Per the pytest changelog:

> [#10196](https://github.com/pytest-dev/pytest/issues/10196):` [PytestReturnNotNoneWarning](https://docs.pytest.org/en/7.2.x/reference/reference.html#pytest.PytestReturnNotNoneWarning) is now a subclass of [PytestRemovedIn8Warning](https://docs.pytest.org/en/7.2.x/reference/reference.html#pytest.PytestRemovedIn8Warning): the plan is to make returning non-None from tests an error in the future.

Reply:
Looks like pytest isn't gonna back down, so we're blocked by https://github.com/matplotlib/pytest-mpl/issues/183 . I see two immediate actions we can take:

* Pin maxversion of pytest to `<7.2` to immediately unblock all the other PRs for feature freeze week. Turn this PR to draft until upstream can figure out the compatibility.
* Continue with this PR but use `pytest.mark.filterwarnings` to ignore this warning for the affected image tests. But this is also risky because if we forget and they turn warning into exception in a future release, we still have to pin maxversion then.

What to do?